content/articles/save-and-load-stacked-ensembles-in-onnx/index.md

```diff
@@ -0,0 +1,256 @@
+```python
+onx = convert_sklearn(pipeline,
+                      initial_types=
+                      initial_types)
+
+with open("stacked_clf.onnx", "wb") as f:
+    f.write(onx.SerializeToString())
+```
+The model is saved sucessfully.
+
+
+***Note:** If establishing the initial types is challenging, for example, the data has too many features, the ```to_onnx``` method handles this directly for you.
+You just need to pass the x test data (or one of it's column) as an argurment and ONNX extracts it automatically.
+
+```python
+# Use a section of data instead of x_test to avoid key errors
+x=data.loc[44].to_numpy(dtype='float32')
+# Give x as a keyword argument by using X=x
+# Note case-sensivity
+onx=skl2onnx.to_onnx(pipeline, X=x)
+
+with open("stacked_clf.onnx", "wb") as f:
+    f.write(onx.SerializeToString())
+```
+
+
+### Loading the model using ONNX runtime Inference Session
+To make predictions from the model, import ```onnxruntime``` and from it, call ```InferenceSession```.
+```python
+import onnxruntime as rt
+sess = rt.InferenceSession("stacked_clf.onnx")
+input_name = sess.get_inputs()[0].name
+label_name = sess.get_outputs()[0].name
+pred_onx = sess.run([label_name], 
+                 {input_name:
+                   x_test.to_numpy(dtype='float32')})
+```
+```x_test``` can be replaced by an array of the shape and type of the test data
+
+Let us see our predictions:
+```python
+pred_onx
+```
+Output:
+```
+[array([0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0,
+        0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0,
+        0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1,
+        0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0,
+        0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1,
+        0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
+        0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
+        0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 0,
+        0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0,
+        0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1,
+        1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
+        1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0], dtype=int64)]
+```        
+###  Conclusion
+In this tutorial, you learnt how to install ONNX and onnxruntime, determine ONNX input initialtypes, serializing and saving a stacked ensemble to ONNX format and finally loading it to production using an Onnx runtime Inference session.
+This model can now be served through a Web App framework like Streamlit and Dash, used in the backend (Django, Flask) or an API.
+In case of a hard nut to crack with ONNX you can raise an issue on [onnx's github](https://github.com/onnx/sklearn-onnx/issues)
+
```

Comment:
Add an URL containing full working code. (It can be colab notebook or Github repo)